src/dbnode/client/session.go

```diff
@@ -3232,20 +3256,32 @@ type streamBlocksResult struct {
 }
 
 func newStreamBlocksResult(
+	nCtx namespace.Context,
 	opts Options,
 	resultOpts result.Options,
 	outputCh chan<- peerBlocksDatapoint,
 	tagDecoder serialize.TagDecoder,
 	idPool ident.Pool,
 ) *streamBlocksResult {
 	return &streamBlocksResult{
-		baseBlocksResult: newBaseBlocksResult(opts, resultOpts),
+		baseBlocksResult: newBaseBlocksResult(nCtx, opts, resultOpts),
 		outputCh:         outputCh,
 		tagDecoder:       tagDecoder,
 		idPool:           idPool,
 	}
 }
 
+func getContextFor(id ident.ID, opts Options) namespace.Context {
+	schemaReg := opts.SchemaRegistry()
+	if id == nil || schemaReg == nil {
```

Comment:
Why not error here?

Reply:
This is to make it backward compatible for existing code (m3tsz) and test code where schema registry is not populated and/or namespace id is set in test setup. 

I haven't written any new test yet (to test this feature). For existing test, I am still thinking about how we should proceed with proto encoding/schema. One way could be to run another test target that have proto encoder/schema enabled by default, this way, we can catch any schema issue earlier.

Reply:
Can we make this explicit - separate function without the args or an enum/sentinel to say something like what mode we’re in? Trying to avoid having nil mean anything 

Would be better to rework the tests to do the right thing instead of changing production code to make it easier to accommodate tests.

Reply:
I removed the check.

Reply:
considering this is outdated and the new place no longer check for nil. The caller should make sure it is not nil, o/w it is a program error we should panic/npe asap. 

I will resolve this. @prateek, take a look at namespace/context.go for the new function: NewContextFor